sdc/src/util/json.d

```diff
@@ -238,22 +238,23 @@ public:
      */
     @trusted pure nothrow this(bool boolean) inout {
         _boolean = boolean;
-        _type = JsonType.Bool;
+        _type = JSON_TYPE.BOOL;
     }
 
     /**
      * Initialize the JSON type from a floating point value.
      */
-    @trusted pure nothrow this(double floating) inout {
+    @trusted pure nothrow this(real floating) inout {
         _floating = floating;
-        _type = JsonType.Float;
+        _type = JSON_TYPE.FLOAT;
     }
 
     /**
      * Initialize the JSON type explicitly to null.
      */
-    @trusted pure nothrow this(typeof(null) nothing = null) inout {
-        _type = JsonType.Null;
+    @trusted pure nothrow this(typeof(null) nothing) inout {
+        _integer = 0;
+        _type = JSON_TYPE.NULL;
```

Comment:
can't have a no-arg constructor. This was the problem I had compiling SDC and was already fixed upstream.